### DISCUSSION TITLE
Rework KinesisSettings

### DISCUSSION THREAD

**Discussion by etspaceman** [laserdisc-io/fs2-aws#976] [Ideas]
Hello - 

`KinesisSettings` seems to be responsible for two things today:

- Basic KCL Configuration, largely used for `ConfigsBuilder`
- The buffer size of the FS2 Queue

I propose separating these notions. This should go a long way to cleaning up the API as well. For example, today we have two primary `create` methods:

- A [schedulerFactory](https://github.com/laserdisc-io/fs2-aws/blob/main/fs2-aws-kinesis/src/main/scala/fs2/aws/kinesis/Kinesis.scala#L117). Intended to allow users to provide much more rich configuration and create the Scheduler themselves
- [AWS Clients](https://github.com/laserdisc-io/fs2-aws/blob/main/fs2-aws-kinesis/src/main/scala/fs2/aws/kinesis/Kinesis.scala#L126). Intended to be a simple way to create consumers, and leverages the `ConfigsBuilder` that the KCL provides

The issue here is if a user elects to use a `schedulerFactory` create method, then they are again forced to provide `KinesisSettings` when running `readFromStream`, most of which the configuration was already provided in the `schedulerFactory` setup. It is cumbersome.

Separating these notions would allow users to create a consumer via a `schedulerFactory`, and only have to provide the queue's buffer size.

**Comment by semenodm** [comment]
@etspaceman can you check the #1030
